packages/doenetml-worker-rust/lib-doenetml-core/src/core/document_model/prop_calculation.rs

```diff
@@ -320,43 +322,94 @@ impl DocumentModel {
+                                        content_refs.push(ContentRef::Component(node.component_idx().into()));
+                                    }
+                                    GraphNode::String(_) => {
+                                        content_refs.push(ContentRef::String(node.idx().into()));
+                                    }
+                                    GraphNode::Prop(_) => {
+                                        // The referent of a PropValue::ContentRef child should be forwarded.
+                                        // Note: the filter is *not* applied to the forward referent.
+                                        let c_refs: ContentRefs = self._get_prop_unchecked(node, query_node).value.try_into().unwrap();
+                                        content_refs.extend(c_refs.into_vec());
+                                    }
+                                    _ => panic!(
+                                        "Unexpected child of `GraphNode::Query` coming from `DataQuery::ComponentRefs`. Got node `{:?}`",
+                                        node
+                                    ),
+                                }
                             }
-                            GraphNode::Prop(_) => {
-                                // The referent of a PropValue::ContentRef child should be forwarded.
-                                // Note: the filter is *not* applied to the forward referent.
-                                let c_refs: ContentRefs = self._get_prop_unchecked(node, query_node).value.try_into().unwrap();
-                                content_refs.extend(c_refs.into_vec());
-                            }
-                            _ => panic!(
-                                "Unexpected child of `GraphNode::Query` coming from `DataQuery::ComponentRefs`. Got node `{:?}`",
-                                node
-                            ),
+                        }
+
+                        DataQueryResult {
+                            values: vec![PropWithMeta {
+                                value: PropValue::ContentRefs(Rc::new(content_refs.into())),
+                                came_from_default: false,
+                                changed: true,
+                                origin: Some(query_node),
+                            }],
                         }
                     }
-                }
+                    // `AnnotatedComponentRefs` are very similar to `ComponentRefs`, but they also include
+                    // information about whether the content is original or comes from extending some other element.
+                    DataQuery::AnnotatedComponentRefs { .. } => {
+                        let content_children =
+                            self.get_component_content_children_annotated(component_idx);
+
+                        let mut content_refs: Vec<(ContentRef, ElementRefAnnotation)> = Vec::new();
```

Comment:
I'd like another name than `content_refs` to clarify that this is more. (I was initially confused how you could do a `content_refs.into()` to get an `AnnotatedContentRefs`.) Maybe something like `content_refs_with_annotations`?